src/main/resources/config/liquibase/changelog/20241107130000_changelog.xml

```diff
@@ -0,0 +1,66 @@
+    <changeSet id="20241107130000" author="stoehrj">
+        <createTable tableName="learner_profile">
+            <column name="id" type="BIGINT" autoIncrement="true">
+                <constraints primaryKey="true" nullable="false"/>
+            </column>
+        </createTable>
+
+        <addColumn tableName="jhi_user">
+            <column name="learner_profile_id" type="bigint"/>
+        </addColumn>
+
+        <addForeignKeyConstraint baseColumnNames="learner_profile_id" baseTableName="jhi_user" constraintName="FK_user_learner_profile" deferrable="false" initiallyDeferred="false" referencedColumnNames="id" referencedTableName="learner_profile" validate="true"/>
+
+        <createIndex tableName="jhi_user" indexName="idx_user_learner_profile">
+            <column name="learner_profile_id"/>
+        </createIndex>
+
+        <sql>
+            INSERT INTO learner_profile (id)
+            SELECT u.id
+            FROM jhi_user u;
+        </sql>
+
+        <sql>
+            UPDATE jhi_user u
+            SET learner_profile_id = u.id;
+        </sql>
+
+        <createTable tableName="course_learner_profile">
+            <column name="id" type="BIGINT" autoIncrement="true">
+                <constraints primaryKey="true" nullable="false"/>
+            </column>
+            <column name="aim_for_grade_or_bonus" type="int"/>
+            <column name="time_investment" type="int"/>
+            <column name="repetition_intensity" type="int"/>
+            <column name="learner_profile_id" type="bigint">
+                <constraints nullable="false"/>
+            </column>
+            <column name="course_id" type="bigint">
+                <constraints nullable="false"/>
+            </column>
+        </createTable>
+
+        <addForeignKeyConstraint baseColumnNames="learner_profile_id" baseTableName="course_learner_profile" constraintName="FK_course_learner_profile_learner_profile" deferrable="false" initiallyDeferred="false" referencedColumnNames="id" referencedTableName="learner_profile" validate="true"/>
+        <addForeignKeyConstraint baseColumnNames="course_id" baseTableName="course_learner_profile" constraintName="FK_course_learner_profile_course" deferrable="false" initiallyDeferred="false" referencedColumnNames="id" referencedTableName="course" validate="true"/>
+
+        <createIndex tableName="course_learner_profile" indexName="idx_clp_learner_profile">
+            <column name="learner_profile_id"/>
+        </createIndex>
+        <createIndex tableName="course_learner_profile" indexName="idx_clp_course">
+            <column name="course_id"/>
+        </createIndex>
+
+
+        <sql>
+            INSERT INTO course_learner_profile (learner_profile_id, course_id)
+            SELECT lp.id, c.id
+            FROM learner_profile lp, course c
+            WHERE c.semester = 'WS24/25' AND c.test_course = FALSE AND c.learning_paths_enabled = TRUE;
```

Comment:
This migration is valid for our instance. If other institutions migrate only next semester it'll not be handled correctly I think.